lsls/lsls.c

```diff
@@ -1,18 +1,63 @@
 #include <stdio.h>
-#include <dirent.h>
+#include <dirent.h> //includes DIR, struct dirent, opendir(), readdir(), and closedir()
+#include <sys/stat.h> //includes stat(), st_size, st_mode
 
 /**
  * Main
  */
 int main(int argc, char **argv)
 {
   // Parse command line
+  // Open directory - opendir()
+  // Repeatly read and print entries - readdir()
+  // Close directory - closedir()
 
-  // Open directory
+  DIR *dir;
+  struct dirent *dirent;
+  struct stat buffer;
 
-  // Repeatly read and print entries
+  //Opens Dir
+  if (argc > 2)
+    {
+      printf("Opening directory '%s'.\n", argv[1]);
+      dir = opendir(argv[1]);
+    }
+  else if (argc < 2)
+  {
+    printf("A directory was not specified, printing content from '.' directory.\n");
+    dir = opendir(".");
+  }
+  else
+  {
+    if (dir == NULL)
+    {
+      printf("Cannot open directory '%s'.\n", argv[1]);
+      exit(1);
```

Comment:
I noticed you were also getting an error for this exit() call. I'm not sure of the exact reason for it, but you stop this warning by return doing  `return 1` instead of `exit(1)`